boards/risc-v/esp32c6/common/src/esp_board_oa_tc6.c

```diff
@@ -0,0 +1,212 @@
+ *   from the MAC-PHY.
+ *
+ * Input Parameters:
+ *   config  - reference to the MAC-PHY configuration
+ *   handler - the interrupt handler function
+ *   arg     - reference passed to the handler on invocation
+ *
+ * Returned Value:
+ *   On success OK is returned, otherwise ERROR is returned.
+ *
+ ****************************************************************************/
+
+static int board_oa_tc6_attach(FAR const struct oa_tc6_config_s *config,
+                               xcpt_t handler, FAR void *arg)
+{
+  int pin;
+
+  switch (config->id)
+    {
+      case SPIDEV_ETHERNET(0):
+          pin = BOARD_OA_TC6_0_INT_PIN;
+          break;
+      default:
+
+          /* Unknown id */
+
+          DEBUGPANIC();
+          return ERROR;
+    }
+
+  esp_configgpio(pin, INPUT_FUNCTION_2 | PULLUP);
+  irq_attach(ESP_PIN2IRQ(pin), handler, arg);
+
+  return OK;
+}
+
+/****************************************************************************
+ * Name: board_oa_tc6_enable
+ *
+ * Description:
+ *   Enable or disable the interrupt on the hardware level.
+ *   Enable when the enable argument is true, otherwise disable.
+ *
+ * Input Parameters:
+ *   config - reference to the MAC-PHY configuration
+ *   enable - enable / disable control
+ *
+ * Returned Value:
+ *   On success OK is returned, otherwise ERROR is returned.
+ *
+ ****************************************************************************/
+
+static int board_oa_tc6_enable(FAR const struct oa_tc6_config_s *config,
+                               bool enable)
+{
+  int pin;
+
+  switch (config->id)
+    {
+      case SPIDEV_ETHERNET(0):
+          pin = BOARD_OA_TC6_0_INT_PIN;
+          break;
+      default:
+
+          /* Unknown id */
+
+          DEBUGPANIC();
+          return ERROR;
+    }
+
+  if (enable)
+    {
+      esp_gpioirqenable(ESP_PIN2IRQ(pin), FALLING);
+    }
+  else
+    {
+      esp_gpioirqdisable(ESP_PIN2IRQ(pin));
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: board_oa_tc6_initialize
+ *
+ * Description:
+ *   Initialize and register the OA-TC6 10BASE-T1x network driver.
+ *
+ * Input Parameters:
+ *   None
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int board_oa_tc6_initialize(void)
+{
+  FAR struct spi_dev_s *spi;
+  int ret;
+
+  spi = esp_spibus_initialize(ESPRESSIF_SPI2);
+  if (!spi)
+    {
+      syslog(LOG_ERR,
+             "ERROR: Failed to initialize SPI port %d\n", ESPRESSIF_SPI2);
+      return -ENODEV;
+    }
+
+  /* Bind the SPI port and config to the OA-TC6 driver */
+
+  ret = oa_tc6_initialize(spi, &g_esp_oa_tc6_config);
+  if (ret < 0)
```

Comment:
```suggestion
  if (ret != OK)
```
Reason for this is that I noticed some return codes coming with ERROR but not negative.